server/src/main/scala/org/scassandra/server/cqlmessages/CqlProtocolHelper.scala

```diff
@@ -43,116 +48,85 @@ object CqlProtocolHelper {
     bs.result().toArray
   }
 
+  /**
+   * Serializes given [[Byte]] into [byte] bytes.
+   */
   def serializeByte(byte : Byte) : Array[Byte] = {
     Array(byte)
   }
 
+  /**
+   * Serializes given [[Int]] into [int] bytes.
+   */
   def serializeInt(int: Int) : Array[Byte] = {
     val frameBuilder = ByteString.newBuilder
     frameBuilder.putInt(int)
     frameBuilder.result().toArray
   }
 
+  /**
+   * Serializes given [[Short]] into [short] bytes.
+   */
   def serializeShort(short : Short) : Array[Byte] = {
     val frameBuilder = ByteString.newBuilder
     frameBuilder.putShort(short)
     frameBuilder.result().toArray
   }
 
-  def serializeBooleanValue(bool : Boolean) : Array[Byte] = {
-    Array[Byte](0,0,0,1, if (bool) 1 else 0)
-  }
-
-  def serializeBigIntValue(value : Long) : Array[Byte] = {
-    val frameBuilder = ByteString.newBuilder
-    frameBuilder.putInt(8)
-    frameBuilder.putLong(value)
-    frameBuilder.result().toArray
-  }
-
-  def serializeDecimalValue(decimal: java.math.BigDecimal) : Array[Byte] = {
-    val scale = decimal.scale()
-    val unscaledValue = decimal.unscaledValue().toByteArray
-    val length = 4 + unscaledValue.length
+  /**
+   * Serializes the given collection into a byte array using the given serializer for encoding
+   * individual elements and size function for writing size of collection.
+   * @param sizeF Function used for writing number of collection elements.
+   * @param input Collection to serialize.
+   * @param entrySerializer Serializer function that accepts a value and encodes it into bytes.
+   * @tparam U type of elements being encoded.
+   * @return The encoded bytes.
+   */
+  def serializeCollection[U](sizeF: (ByteStringBuilder, Int) => ByteStringBuilder)(input: TraversableOnce[U], entrySerializer: U => Array[Byte]) : Array[Byte] = {
```

Comment:
Do you think we should attach the sizeF to either the ProtocolVersion class or something? No problem is this a pain and don't have it here. Isolating the differences will become more important the more we add